hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMAllocateBlockRequest.java

```diff
@@ -176,12 +177,18 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
       // write ACL on key. Add client id to key name if ozone native
       // authorizer is configured.
       Configuration config = ozoneManager.getConfiguration();
-      String keyNameForAclCheck = keyName;
       if (OmUtils.isNativeAuthorizerEnabled(config)) {
-        keyNameForAclCheck = keyName + "/" + allocateBlockRequest.getClientID();
+        String keyNameForAclCheck =
+            keyName + "/" + allocateBlockRequest.getClientID();
+        // During allocate block request, it is possible that key is
+        // not present in the key table and hence setting the resource type
```

Comment:
Can we move this to a common method, as it is being used in KeyCommitRequest also.